pkg/provider/ip/provider.go

```diff
@@ -339,7 +339,7 @@ func (p *ipv4Provider) ReSyncPool(job *worker.WarmPoolJob) {
 func (p *ipv4Provider) DeletePrivateIPv4AndUpdatePool(job *worker.WarmPoolJob) {
 	instanceResource, found := p.getInstanceProviderAndPool(job.NodeName)
 	if !found {
-		p.log.Error(fmt.Errorf("cannot find the instance provider and pool form the cache"), "node", job.NodeName)
+		p.log.Error(utils.ErrNotFound, "cannot find the instance provider and pool from the cache", "node", job.NodeName)
```

Comment:
Should we define this error in utils as it is used in multiple statements? 
Also would be good to keep the node name key consistent, "nodename" should work.

Reply:
Hi @sushrk, `utils.ErrNotFound` is already defined in utils. Are you referring to the message `"cannot find the instance provider and pool from the cache"`? I can move it to the same `errors.go` if that works.

Reply:
Thanks for the suggestion. I've updated both.